lib/nerves_runtime/kv.ex

```diff
@@ -275,28 +260,56 @@ defmodule Nerves.Runtime.KV do
     {:reply, reply, s}
   end
 
-  defp active(s), do: Map.get(s, "nerves_fw_active", "")
+  defp active(s), do: Map.get(s.contents, "nerves_fw_active", "")
 
   defp active(key, s) do
-    Map.get(s, "#{active(s)}.#{key}")
+    Map.get(s.contents, "#{active(s)}.#{key}")
   end
 
   defp filter_trim_active(s, active) do
-    Enum.filter(s, fn {k, _} ->
+    Enum.filter(s.contents, fn {k, _} ->
       String.starts_with?(k, active)
     end)
     |> Enum.map(fn {k, v} -> {String.replace_leading(k, active, ""), v} end)
     |> Enum.into(%{})
   end
 
   defp do_put(kv, s) do
-    case mod().put(kv) do
-      :ok -> {:ok, Map.merge(s, kv)}
+    case s.backend.save(kv, s.options) do
+      :ok -> {:ok, %{s | contents: Map.merge(s.contents, kv)}}
       error -> {error, s}
     end
   end
 
-  defp mod() do
-    Application.get_env(:nerves_runtime, :modules)[__MODULE__] || @default_mod
+  defguardp is_module(v) when is_atom(v) and not is_nil(v)
+
+  defp initial_state(options) do
+    case options[:kv_backend] do
+      {backend, opts} when is_module(backend) and is_list(opts) ->
+        initialize(backend, opts)
+
+      backend when is_module(backend) ->
+        initialize(backend, [])
+
+      _ ->
+        # Handle Nerves.Runtime v0.12.0 and earlier way
+        initial_contents = options[:modules][Nerves.Runtime.KV.Mock]
+        initialize(Nerves.Runtime.KVBackend.InMemory, contents: initial_contents)
+    end
+  rescue
+    error ->
+      Logger.error("Nerves.Runtime has a bad KV configuration: #{inspect(error)}")
+      initialize(Nerves.Runtime.KVBackend.InMemory, [])
+  end
+
+  defp initialize(backend, options) do
+    case backend.load(options) do
+      {:ok, state} ->
```

Comment:
```suggestion
      {:ok, contents} ->
```
More of a bit, but there is a few mismatches of naming around this "state" vs "contents" (here, in the KVBackend module doc and type specs, etc)

I think we should pick one representation and use that in each reference spot. Since the module doc says "load KV store and return its contents", I might vote we stick with "contents"

Reply:
Thanks for noticing that. I forgot to go back and fix all of those. Totally agree.